File: src/main/java/com/gmail/borlandlp/minigamesdtools/arena/customhandlers/Handler.java
package com.gmail.borlandlp.minigamesdtools.arena.customhandlers;

import com.gmail.borlandlp.minigamesdtools.MinigamesDTools;
import org.bukkit.scheduler.BukkitRunnable;
import org.bukkit.scheduler.BukkitTask;

public abstract class Handler {
    private BukkitTask task;
    private int delay = 0;
    private int period = 20;

    void start() {
        final Handler task = this;
        this.task = new BukkitRunnable() {
            public void run() {
                task.doWork();
            }
        }.runTaskTimer(MinigamesDTools.Companion.getInstance(), this.delay, this.period);
    }

    void stop() {
        if(this.task != null) {
            this.task.cancel();
        }
    }

    public abstract void doWork();
}

File: src/main/kotlin/com/gmail/borlandlp/minigamesdtools/arena/customhandlers/Handler.kt
package com.gmail.borlandlp.minigamesdtools.arena.customhandlers

import com.gmail.borlandlp.minigamesdtools.MinigamesDTools.Companion.instance
import org.bukkit.scheduler.BukkitRunnable
import org.bukkit.scheduler.BukkitTask

abstract class Handler {
    private var task: BukkitTask? = null
    private val delay = 0
    private val period = 20

    fun start() {
        val task = this
        this.task = object : BukkitRunnable() {
            override fun run() {
                task.doWork()
            }
        }.runTaskTimer(instance, delay.toLong(), period.toLong())
    }

    fun stop() {
        task?.cancel()
    }

    abstract fun doWork()
}